### PR TITLE
Svelte: Support v5 prereleases

### LINKED ISSUE #24886: [Bug]: Instantiating a component with `new` is no longer valid in Svelte 5.
### Describe the bug

Just created an empty SvelteKit Library project with Beta version of Svelte 5 enabled and installed Storybook.

This is what i get when viewing demo component stories:

Zrzut ekranu 2023-11-17 o 11 48 09

### To Reproduce

```
pnpm create svelte
```

Choose Svelte 5 beta.

```
pnpm dlx storybook@latest init
```

### System

_No response_

### Additional context

I managed to find that this particular code causes the error:

https://github.com/storybookjs/storybook/blob/main/code/renderers/svelte/src/render.ts#L69

According to [docs](https://svelte-5-preview.vercel.app/docs/breaking-changes#components-are-no-longer-classes), this:

```js
let createdComponent = new PreviewRender({ target: canvasElement, props: { storyFn, storyContext, name, kind, showError } });
```

should be changed to this:

```js
import { mount } from 'svelte';

function renderToCanvas({ storyFn, kind, name, showMain, showError, storyContext, forceRemount }, canvasElement) {

  // Rest of the code

  let createdComponent = mount(PreviewRender, { target: canvasElement, props: { storyFn, storyContext, name, kind, showError } });

  // Rest of the code
}

```

### PR DESCRIPTION
**Pull request by allozaur**
Closes #24886

## What I did

Changes for rendering stories for Svelte 5 where ["Components are no longer classes"](https://svelte-5-preview.vercel.app/docs/breaking-changes#components-are-no-longer-classes)

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version [`0.0.0-pr-24889-sha-7abeedf4`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24889-sha-7abeedf4). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-24889-sha-7abeedf4`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24889-sha-7abeedf4) |
| **Triggered by** | @JReinhold |
| **Repository** | [allozaur/storybook](https://github.com/allozaur/storybook) |
| **Branch** | [`allozaur/#24886`](https://github.com/allozaur/storybook/tree/allozaur/#24886) |
| **Commit** | [`7abeedf4`](https://github.com/allozaur/storybook/commit/7abeedf44e256dd4d2a7a427157363a875ef9124) |
| **Datetime** | Thu Nov 30 09:30:10 UTC 2023 (`1701336610`) |
| **Workflow run** | [7044517732](https://github.com/storybookjs/storybook/actions/runs/7044517732) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=24889`_
</details>

### CODE REVIEW

**Comment by JReinhold** [comment]
We're running into this issue currently: https://github.com/sveltejs/svelte/issues/9716

It means that when you change a control, any of the controls that are unset will make the component discard the fallback value and turn `undefined` instead.

**Comment by JReinhold** [comment]
I've added Svelte 4 support back, because Svelte 5 still seems a bit rough around the edges, so I don't want to force the use of it in SB 8.0 prereleases just yet. Later in the release cycle we might remove Svelte 4 support again, TBD.

I've added a new SvelteKit prerelease sandbox that uses Svelte 5.

Generating sandboxes here: https://github.com/storybookjs/storybook/actions/runs/7046672852/job/19178935601?pr=24889

**Review by yannbf — APPROVED** [review]
Looking good! But the sandboxes should be generated in a separate branch that is not a fork

**Review by JReinhold — APPROVED** [review]

**Comment by JReinhold** [comment]
Sandbox is failing to be created because a transitive upstream package doesn't allow Svelte 5 yet: https://github.com/sveltejs/svelte-hmr/pull/87

File: scripts/tasks/sandbox-parts.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const install: Task['run'] = async ({ sandboxDir }, { link, dryRun, debug
     // of any storybook packages as verdaccio is not able to both proxy to npm and publish over
     // the top. In theory this could mask issues where different versions cause problems.
     await addPackageResolutions({ cwd, dryRun, debug });
-    await configureYarn2ForVerdaccio({ cwd, dryRun, debug });
+    await configureYarn2ForVerdaccio({ cwd, dryRun, debug, sandboxDir });
```
Comment (valentinpalkovic) [inline scripts/tasks/sandbox-parts.ts:96]: Isn't `cwd === sandboxDir`?

Comment (valentinpalkovic) [reply scripts/tasks/sandbox-parts.ts:96]: Also, I think using `key` here is more robust than relying on a dir name. Key can then also be typed properly

Comment (JReinhold) [reply scripts/tasks/sandbox-parts.ts]: you're right, I was copy-pasting too much, `key` is much better.

**Pull request merged**